### PR TITLE
Verify mask fit with reference in get_data_as_mask

### PR DESCRIPTION
**Pull request by EmmaRenauld**
# Quick description

I was seeing many times the same line:

```
if args.mask is None:
    mask = None
else:
   mask = get_data_as_mask()
   verify_shape
```

Replaced by single-line with verification inside get_data_as_mask. Makes the verification more thorough (was not done in all scripts) and will help with coverage.

EDIT: 

After discussion with Arnaud, instead, not verifying shape there, but using assert_headers_compatible at the beginning.

+ added missing mask in many `assert_inputs_exit` calls.
+ added missing reference in many `assert_inputs_exit` calls.
...

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [ ] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [ ] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scripts/scil_tractogram_uniformize_endpoints.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def main():
     args = parser.parse_args()
     logging.getLogger().setLevel(logging.getLevelName(args.verbose))
 
-    assert_inputs_exist(parser, args.in_bundle)
+    assert_inputs_exist(parser, args.in_bundle, args.reference)
```
Comment (arnaudbore) [inline scripts/scil_tractogram_uniformize_endpoints.py:80]: assert_headers_compatible with target_roi and reference

**Review by arnaudbore — COMMENTED** [review]
76 files always a pleasure :) Great PR everything make sense. A couple of comment, @karanphil will check it before moving forward.

File: scripts/scil_tracking_pft_maps_edit.py [hunk 2]
```diff
@@ -14,10 +14,9 @@
 import numpy as np
 
 from scilpy.io.image import get_data_as_mask
-from scilpy.io.utils import (add_overwrite_arg,
-                             add_verbose_arg,
-                             assert_inputs_exist,
-                             assert_outputs_exist)
+from scilpy.io.utils import (add_overwrite_arg, add_verbose_arg,
+                             assert_inputs_exist, assert_outputs_exist,
+                             assert_headers_compatible)
```
Comment (karanphil) [inline scripts/scil_tracking_pft_maps_edit.py:19]: This is what I mean by "two columns" :)

**Review by karanphil — COMMENTED** [review]
Really nice addition! I found a few scripts where I don't understand why we don't use your function `assert_headers_compatible`, especially when we have multiple tractograms (in scil_tractogram or scil_bundle). I also added a few esthetic comments. Moreover, I think some scripts might have been left out, like `scil_dwi_concatenate.py`, where a check is done with `is_header_compatible` but could be replaced by your function.

Good job!

**Comment by EmmaRenauld** [comment]
> Really nice addition! I found a few scripts where I don't understand why we don't use your function `assert_headers_compatible`, especially when we have multiple tractograms (in scil_tractogram or scil_bundle). I also added a few esthetic comments. Moreover, I think some scripts might have been left out, like `scil_dwi_concatenate.py`, where a check is done with `is_header_compatible` but could be replaced by your function.
> 
> Good job!

Yeah, at first I was managing only script calling the function get_data_as_mask. Then I also managed all scripts using add_reference_arg. But I didn't go through all scripts exhaustively. Should I? In a new PR?

File: scilpy/io/utils.py [hunk 3]
```diff
@@ -713,17 +712,32 @@ def is_header_compatible_multiple_files(parser, list_files,
     ---------
     parser: argument parser
         Will raise an error if a file is not compatible.
-    list_files: List[str]
+    required: List[str]
         List of files to test
+    optional: List[str or None]
+        List of files. May contain None, they will be discarted.
     verbose_all_compatible: bool
```
[the remark this answers is not in the document]
Comment (EmmaRenauld) [reply scilpy/io/utils.py:719]: That variable is only used in the script scil_header_validate_compatibility.py. You can check and tell me how you'd like to handle it.

File: scilpy/io/utils.py [hunk 4]
```diff
@@ -713,17 +712,32 @@ def is_header_compatible_multiple_files(parser, list_files,
...
         If true will print a message when everything is okay
     reference: str
```
[the remark this answers is not in the document]
Comment (EmmaRenauld) [reply scilpy/io/utils.py:721]: Considering that the typical argparser name is --reference, and not --reference_tck, I would keep it as is.

File: scripts/scil_dki_metrics.py [hunk 5]
```diff
@@ -59,7 +59,8 @@
 from scilpy.io.image import get_data_as_mask
 from scilpy.io.utils import (add_overwrite_arg, add_skip_b0_check_arg,
                              add_verbose_arg, assert_inputs_exist,
-                             assert_outputs_exist, add_tolerance_arg, )
+                             assert_outputs_exist, add_tolerance_arg,
+                             assert_headers_compatible, )
```
[the remark this answers is not in the document]
Comment (EmmaRenauld) [reply scripts/scil_dki_metrics.py:63]: oups ;)

File: scripts/scil_bundle_score_same_bundle_many_segmentations.py [hunk 6]
```diff
@@ -217,9 +219,8 @@ def main():
     else:
         gs_binary_3d = get_data_as_mask(nib.load(args.voxels_measures[0]))
         gs_binary_3d[gs_binary_3d > 0] = 1
-        tracking_mask_data = get_data_as_mask(nib.load(
-                                                args.voxels_measures[1]))
-        tracking_mask_data[tracking_mask_data > 0] = 1
```
Comment (arnaudbore) [inline scripts/scil_bundle_score_same_bundle_many_segmentations.py:222]: why did you remove this ? useless ?

**Review by arnaudbore — COMMENTED** [review]
a quick question otherwise we are getting there !

File: scripts/scil_bundle_score_same_bundle_many_segmentations.py (hunk 6 above)
Comment (karanphil) [reply scripts/scil_bundle_score_same_bundle_many_segmentations.py:222]: The get_data_as_mask function now covers for that.

**Review by karanphil — APPROVED** [review]
There are still a few scripts were the imports are not "cute" (two columns), but I guess it's just a detail! GTG

File: scripts/scil_bundle_score_same_bundle_many_segmentations.py (hunk 6 above)
Comment (karanphil) [reply scripts/scil_bundle_score_same_bundle_many_segmentations.py:222]: The get_data_as_mask function now covers for that.
